language/vm/vm_runtime/src/code_cache/module_cache.rs

```diff
@@ -284,21 +287,29 @@ impl<'alloc> VMModuleCache<'alloc> {
         }
         let def = {
             let struct_def = module.struct_def_at(idx);
-            let mut field_types = vec![];
-            for field in module.field_def_range(struct_def.field_count, struct_def.fields) {
-                let ty = try_runtime!(self.resolve_signature_token_with_fetcher(
-                    module,
-                    &module.type_signature_at(field.signature).0,
-                    gas_meter,
-                    fetcher
-                ));
-                if let Some(t) = ty {
-                    field_types.push(t);
-                } else {
-                    return Ok(Ok(None));
+            match &struct_def.field_information {
+                StructFieldInformation::Native => return Err(VMInvariantViolation::LinkerError),
```

Comment:
TODO for handling this in the future or returning a more descriptive error?

Reply:
It's the same one we use for native procedures, but I'll add a TODO